eng/breakingChanges/breaking-change-doc.ps1

```diff
@@ -0,0 +1,1231 @@
+
+**Generated**: $(Get-Date -Format 'yyyy-MM-dd HH:mm:ss')
+**Mode**: $queryInfo
+
+## Summary
+
+- **Total PRs collected**: $($analysisData.Count)
+- **PRs with existing docs issues**: $($analysisData | Where-Object HasDocsIssue | Measure-Object).Count
+- **PRs needing docs issues**: $($analysisData | Where-Object { -not $_.HasDocsIssue } | Measure-Object).Count
+
+## PRs Needing Documentation
+
+$($analysisData | Where-Object { -not $_.HasDocsIssue } | ForEach-Object {
+    "- **PR #$($_.Number)**: $($_.Title)`n  - URL: $($_.Url)`n  - Feature Areas: $($_.FeatureAreas)`n"
+} | Out-String)
+
+## PRs With Existing Documentation
+
+$($analysisData | Where-Object HasDocsIssue | ForEach-Object {
+    "- **PR #$($_.Number)**: $($_.Title)`n  - URL: $($_.Url)`n"
+} | Out-String)
+"@
+
+$summaryReport | Out-File (Join-Path $dataDir "summary_report.md") -Encoding UTF8
+
+Write-Host "✅ Data collection completed" -ForegroundColor Green
+Write-Host "   📊 Summary: $(Join-Path $dataDir "summary_report.md")"
+Write-Host "   📋 Combined: $(Join-Path $dataDir "combined.json")"
+Write-Host "   📄 Individual: $(Join-Path $dataDir "pr_*.json") ($($analysisData.Count) files)"
+
+if ($CollectOnly) {
+    exit 0
+}
+
+# STEP 2: Generate breaking change issues
+Write-Host "`n📝 Step 2: Generating breaking change documentation..." -ForegroundColor Green
+
+$prsNeedingDocs = $analysisData | Where-Object { -not $_.HasDocsIssue }
+
+if ($prsNeedingDocs.Count -eq 0) {
+    if ($PrNumber) {
+        Write-Host "   PR #$PrNumber already has documentation or doesn't need it." -ForegroundColor Yellow
+    } else {
+        Write-Host "   No PRs found that need documentation issues." -ForegroundColor Yellow
+    }
+    exit 0
+}
+
+if ($PrNumber) {
+    Write-Host "   Processing PR #$PrNumber for issue generation"
+} else {
+    Write-Host "   Processing $($prsNeedingDocs.Count) PRs for issue generation"
+}
+
+foreach ($pr in $prsNeedingDocs) {
+    Write-Host "   🔍 Processing PR #$($pr.Number): $($pr.Title)" -ForegroundColor Cyan
+
+    # Use commits data already collected in Step 1
+    $commits = $pr.Commits
+
+    # Prepare data for LLM
+    $comments = if ($pr.Comments -and $pr.Comments.Count -gt 0) {
+        ($pr.Comments | ForEach-Object { "**@$($_.author.login)**: $(Limit-Text -text $_.body -maxLength 300)" }) -join "`n`n"
+    } else { "No comments" }
+
+    $reviews = if ($pr.Reviews -and $pr.Reviews.Count -gt 0) {
+        ($pr.Reviews | ForEach-Object { "**@$($_.author.login)** ($($_.state)): $(Limit-Text -text $_.body -maxLength 200)" }) -join "`n`n"
+    } else { "No reviews" }
+
+    $closingIssuesInfo = if ($pr.ClosingIssues -and $pr.ClosingIssues.Count -gt 0) {
+        $issuesList = $pr.ClosingIssues | ForEach-Object {
+            $issueComments = if ($_.Comments -and $_.Comments.Count -gt 0) {
+                "Comments: $($_.Comments.Count) comments available"
+            } else {
+                "No comments"
+            }
```

Comment:
We recently hit upon a gap in our processes where we can have both issues and PRs labeled as https://github.com/dotnet/runtime/labels/breaking-change, but we missed auditing breaking change issues to ensure all closing PRs for them were also labeled as breaking.

I don't think there's anything to do _in this PR_ for that, but seeing the traversal of `ClosingIssues` here has me wondering if we should refine our approach for labeling--perhaps to automatically label merged PRs as breaking changes if a breaking-change issue was closed by the PR.

Reply:
It would have to run whenever a label is applied to the issue - make it applied to the corresponding PR(s) whether open/closed.  If we can have the automation do that reliably it could fill that gap.

Another thing we could do would be to nag people when applying the label to an issue and tell them to apply to PR instead.